File: fix-arabic.js
const fs = require('fs');
const path = require('path');

const gamesDir = 'C:\\Users\\Mr Mahmoud Elziadi\\Documents\\GitHub\\Englishtoon\\games';

function getHtmlFiles(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(getHtmlFiles(file));
        } else if (file.endsWith('.html')) {
            results.push(file);
        }
    });
    return results;
}

const files = getHtmlFiles(gamesDir);
console.log(`Fixing Arabic text in ${files.length} files...`);

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let originalContent = content;

    // 1. Fix guard script alert
    content = content.replace(/alert\('O1O.*?U<\.'\);/g, "alert('عذراً لا يمكنك الدخول للعبة مباشرة. يرجى تسجيل الدخول أولاً.');");

    // 2. Fix Score and Lives
    content = content.replace(/<div>\?+:\s*<span id="score"/g, '<div>النقاط: <span id="score"');
    content = content.replace(/<div>\?+:\s*<span id="lives"/g, '<div>القلوب: <span id="lives"');

    // 3. Fix Game Over Screen
    content = content.replace(/(<h1[^>]*>)\?+ \?+!(<\/h1>)/g, '$1انتهت اللعبة!$2');
    content = content.replace(/(<p[^>]*>)\?+:\s*<span id="final-score"/g, '$1النقاط: <span id="final-score"');

    // 4. Fix Buttons
    content = content.replace(/(onclick="startGame\(\)"[^>]*>)\?+!*(<\/button>)/g, '$1إبدأ اللعب$2');
    content = content.replace(/(onclick="startGame\(\)"[^>]*>)\?+ \?+ \?+(<\/button>)/g, '$1إلعب مرة أخرى$2');

    // 5. Fix Start Screen Title
    content = content.replace(/(id="start-screen"[^>]*>[\s\S]*?<h1[^>]*>)\?+.*?(<\/h1>)/g, '$1مرحباً بك في اللعبة$2');

    // 6. Fix Start Screen Subtitle
    content = content.replace(/(id="start-screen"[^>]*>[\s\S]*?<p[^>]*>)\?+.*?(<\/p>)/g, '$1اضغط على زر إبدأ اللعب لتبدأ المغامرة!$2');

    if (content !== originalContent) {
        fs.writeFileSync(file, content, 'utf8');
    }
});

console.log('Fix complete!');

File: fix-arabic.py
import os
import re

games_dir = r"C:\Users\Mr Mahmoud Elziadi\Documents\GitHub\Englishtoon\games"

def get_html_files(directory):
    html_files = []
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith(".html"):
                html_files.append(os.path.join(root, file))
    return html_files

files = get_html_files(games_dir)
print(f"Fixing Arabic text in {len(files)} files...")

for file in files:
    try:
        with open(file, 'r', encoding='utf-8') as f:
            content = f.read()
    except UnicodeDecodeError:
        print(f"Skipping {file} due to encoding error")
        continue

    original_content = content

    # 1. Fix guard script alert (mojibake)
    content = re.sub(r"alert\('O1O.*?U<\.'\);", "alert('عذراً لا يمكنك الدخول للعبة مباشرة. يرجى تسجيل الدخول أولاً.');", content)

    # 2. Fix Score and Lives
    content = re.sub(r'<div>\?+:\s*<span id="score"', '<div>النقاط: <span id="score"', content)
    content = re.sub(r'<div>\?+:\s*<span id="lives"', '<div>القلوب: <span id="lives"', content)

    # 3. Fix Game Over Screen
    content = re.sub(r'(<h1[^>]*>)\?+ \?+!(</h1>)', r'\1انتهت اللعبة!\2', content)
    content = re.sub(r'(<p[^>]*>)\?+:\s*<span id="final-score"', r'\1النقاط: <span id="final-score"', content)

    # 4. Fix Buttons
    content = re.sub(r'(onclick="startGame\(\)"[^>]*>)\?+!*(</button>)', r'\1إبدأ اللعب\2', content)
    content = re.sub(r'(onclick="startGame\(\)"[^>]*>)\?+ \?+ \?+(</button>)', r'\1إلعب مرة أخرى\2', content)

    # 5. Fix Start Screen Title
    content = re.sub(r'(id="start-screen"[^>]*>[\s\S]*?<h1[^>]*>)\?+.*?(</h1>)', r'\1مرحباً بك في اللعبة\2', content)

    # 6. Fix Start Screen Subtitle
    content = re.sub(r'(id="start-screen"[^>]*>[\s\S]*?<p[^>]*>)\?+.*?(</p>)', r'\1اضغط على زر إبدأ اللعب لتبدأ المغامرة!\2', content)

    if content != original_content:
        with open(file, 'w', encoding='utf-8') as f:
            f.write(content)

print("Fix complete!")
